doc/book/src/user/deprecation.md

```diff
@@ -0,0 +1,67 @@
+Deprecated Features
+===================
+
+In order to support the continuous improvement of zcashd, features are
+periodically deprecated and removed when they have been superseded or are no
+longer useful.  Deprecation follows a 3-stage process:
+
+1. Initially, a feature will be marked as DEPRECATED in the release notes and
+   user-facing documentation, but no other changes are made; the feature
+   continues to be available and function as normal. While features at this
+   stage remain enabled by default, they may be explicitly disabled by
+   specifying `-allowdeprecated=none` on the command line when starting the
+   node, or by including `allowdeprecated=none` as a line in the `zcash.conf`
+   file. 
+2. In the stage of deprecation, the feature will be disabled by default.
+   Disabled features may be reenabled via use of the `-allowdeprecated` flag.
+3. In the third stage, the feature is fully removed and is no longer available.
+
+Features that enter Stage 1 in a particular release will be disabled by default
+after no less than than a single major-version release, or 3 minor-version
+releases, and features are only fully removed after two major-version releases.
```

Comment:
Hmm, that's not the same as the developer doc says. And I'm concerned that this will cause significant churn, as major-version releases are synced with network upgrades, so it means we're removing features on the same boundary as updates that are required for network compatibility.

Reply:
My rationale here is that major-version releases are where we make breaking changes to the protocol, and so it makes sense that feature removal will coincide (sometimes necessarily) with such breaking changes. It also makes for greater predictability: users of our APIs can predict accurately that once a feature is disabled by default at a major release version, that it will be removed in the next major release and they'd better have migrated away from the deprecated feature by then.